src/rules/store/on-function-explicit-return-type.ts

```diff
@@ -16,22 +20,54 @@ export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
       category: 'Possible Errors',
       description: 'On function should have an explicit return type.',
       recommended: 'warn',
+      suggestion: true,
     },
     schema: [],
     messages: {
-      [messageId]:
+      [onFunctionExplicitReturnType]:
         'On functions should have an explicit return type when using arrow functions: `on(action, (state): State => {}`.',
+      [onFunctionExplicitReturnTypeSuggest]: 'Add explicit return type.',
```

Comment:
I think we should mention that we've chosen `State` as return type, but that the user should update it if needed.
What do you think?

Reply:
Cool. What's your suggestion?

Reply:
Ha! that's a good question (and of course it's easy for me to point this out without providing something better) 😅
Something like this?

"Add explicit return type (By default `State`, when the interface is named different you need to manually correct the return type)."

Reply:
What about this? Feel free to apply the suggestion if you agree :)

```suggestion
      [onFunctionExplicitReturnTypeSuggest]: 'Add explicit return type `State` (if the interface is named differently you need to manually correct the return type).',
```